earth_enterprise/src/common/gedbroot/eta_dbroot.cc

```diff
@@ -77,22 +77,21 @@ bool EtaDbroot::DecodeBinary(const std::string &binary,
   key_buffer.append(binary, sizeof(etDbaseRootHeader),
                     etEncoder::kDefaultKey.size());
   if (key_buffer != etEncoder::kDefaultKey) {
-    khSimpleException("Corrupted dbroot: Key has unexpected value.");
+    throw khSimpleException("Corrupted dbroot: Key has unexpected value.");
   }
 
   // Read the header and see if it is a valid binary dbroot file.
   std::string header_buffer;
   header_buffer.append(binary, 0, sizeof(etDbaseRootHeader));
   // Decrypt the entire header
-  etEncoder::Decode(&header_buffer[0], header_buffer.size(),
-                    key_buffer.data(), key_buffer.size());
+  etEncoder::DecodeWithDefaultKey(&header_buffer[0], header_buffer.size());
```

Comment:
I switched this (and the Decode call below) to use the default key because the code verifies that the key matches the default key above.